setup.py

```diff
@@ -166,13 +147,26 @@ def run(self):
     gdal_major_version = int(gdal_version_parts[0])
     gdal_minor_version = int(gdal_version_parts[1])
 
+    if (gdal_major_version, gdal_minor_version) < (3, 1):
+        raise SystemExit(
+            "ERROR: GDAL >= 3.1 is required for fiona. "
+            "Please upgrade GDAL."
+        )
+
```

Comment:
Very useful, as this is otherwise not enforced. By chance, it was working with Ubuntu's [libgdal-dev 3.0.4](https://packages.ubuntu.com/focal/libgdal-dev) without too many issues.